src/Raven.Client/Util/TcpUtils.cs

```diff
@@ -144,5 +144,34 @@ private static TcpClient NewTcpClient(TimeSpan? timeout, bool useIPv6)
             Debug.Assert(tcpClient.Client != null);
             return tcpClient;
         }
+
+        internal static async Task<(TcpClient Client, string Url)> ConnectAsyncWithPriority(TcpConnectionInfo info, TimeSpan? tcpConnectionTimeout)
+        {
+            string selectedUrl = null;
+            TcpClient tcpClient = null;
+            
+            foreach (var url in info.TcpServerUrls)
```

Comment:
We need to either fall back to the just the `Url` if the `TcpServerUrls` are not specified or give a good error here.
Otherwise, we'll get NRE when connecting to an old server.

Reply:
for the subscription you're correct we might have null need to handle it because of the client